src/ur/chatgpt.urs

```diff
@@ -23,7 +23,8 @@ functor Make(M : sig
     structure Chat : sig
         val completions : {Model : model,
                            Messages : list {Role : role,
-                                            Content : string}}
+                                            Content : string},
+                            ResponseFormat: {Type: string}}
```

Comment:
Hmm, as I look at this, it feels a bit off.  Why not just `ResponseFormat: string`?

Also, my general feeling is that fields like this that should really only be one of a handful of values should be enumerations (e.g., `datatype ResponseFormat = AsJson | RegularText` or something).  Is there a reason you went with the more general `string`?

Lastly, I know these files can be super light on documentation, but all the more reason that when we make changes to them, we should add that documentation.  I realize looking at this now that I would have no idea what to pass as the value of `ResponseFormat` here.

Reply:
>  Why not just 'ResponseFormat: string'

So according to the OpenAI documentation, the value has to be a dict such as `{type: "text"}`.

>  ... fields like this that should really only be one of a handful of values should be enumerations (e.g., datatype ResponseFormat = AsJson | RegularText or something)

I will update the code and add more documentations shortly.

Reply:
I just updated the code and have made corresponding changes to the nectry repo as well.

Reply:
> So according to the OpenAI documentation, the value has to be a dict such as `{type: "text"}`

Aha, _this_ needs to be in the code.  That is, if someone comes and looks at this code later, they may ask the natural question "why is it this way", and there's nothing about the structure of the code that naturally answers that.  I would put in a doc line that says something like (feel free to reword):

> The OpenAI documentation (found here: <link>) dictates that response type is a field of a dict with key "type".  There are other values allowed in this dict, but we are only using "type" for now.

I'm a big fan of docs answering the question "why".

Reply:
Sounds good. I just fixed it in my most recent commit.